substrafl/index_generator/base.py

```diff
@@ -129,9 +129,10 @@ def n_samples(self, _n_samples: int):
 
     def __iter__(self) -> "BaseIndexGenerator":
         """Required methods for generators, returns ``self``."""
+        self._reset_counter()
         return self
 
-    @abc.abstractclassmethod
+    @abc.abstractmethod
```

Comment:
abstractclassmethod is deprecated since python3.3 and should be replaced by abstractmethod